### PR TITLE
Remove ccache from CI

### PR DESCRIPTION
**Pull request by mrks**
Just today, I had two CI fails that were "fixed" by deleting the ccache folder and re-running the build. We originally hoped that the new ccache version fixed the issues, but it doesn't.

I'd say we retire it. This might make the CI builds a bit longer, but for me, that sounds preferrable to spuriously failing builds - either during compilation or even during execution.

TODO:
- [x] Check if `--privileged=true` is necessary

### CODE REVIEW

**Comment by Bensk1** [comment]
If this is not fixed by the new ccache version, I would also vote for retiring it. Do we have any idea about how much time it saves per full CI run?

File: README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ To use a different one, call `cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILE
...
+For development, you may want to use [ccache](https://ccache.samba.org/), which reduces the time needed for recompiles significantly. Especially when switching branches, this can reduce the time to recompile from several minutes to one or less. On the downside, we have seen random build failures on our CI server, which is why we do not recommend ccache anymore but merely list it as an option. To use ccache, add `-DCMAKE_CXX_COMPILER_LAUNCHER=ccache` to your cmake call. You will need to [adjust some ccache settings](https://ccache.dev/manual/latest.html#_precompiled_headers) either in your environment variables or in your [ccache config](https://ccache.dev/manual/latest.html#_configuration) so that ccache can handle the precompiled headers. On our CI server, this works for us: `CCACHE_SLOPPINESS=file_macro,pch_defines,time_macros CCACHE_DEPEND=1`.
```
Comment (Bensk1) [inline README.md:66]: On our CI server, this work**ED** for us:

**Comment by mrks** [comment]
Will have a look once we have decided on #2114 and this PR can be fully built.

**Comment by mrks** [comment]
> If this is not fixed by the new ccache version, I would also vote for retiring it. Do we have any idea about how much time it saves per full CI run?

Comparing [a build](https://hyrise-ci.epic-hpi.de/blue/organizations/jenkins/hyrise%2Fhyrise/detail/PR-2127/4/pipeline) from #2127 and [this build](https://hyrise-ci.epic-hpi.de/blue/organizations/jenkins/hyrise%2Fhyrise/detail/PR-2108/15/pipeline), it doesn't seem to make a notable difference.

**Comment by Bensk1** [comment]
KillItWithFire

**Pull request merged**